Harvest.Web/ClientApp/src/Tickets/TicketCreate.tsx

```diff
@@ -42,8 +45,30 @@ export const TicketCreate = () => {
   if (project === undefined) {
     return <div>Loading...</div>;
   }
+
+  
+  const checkTicketValidity = async (inputs: any) => {
+    try {
+      await ticketSchema.validate(inputs, { abortEarly: false });
+    } catch (err) {
+      if (err instanceof ValidationError) {
+        return err.errors;
+      }
+    }
+  };
+
   const create = async () => {
-    // TODO: validation
+
+    const ticketErrors = await checkTicketValidity(ticket);
+
+    if (ticketErrors) {
+      if (ticketErrors?.length > 0) {
+        setInputErrors(ticketErrors);
+        return;
+      } else {
+        setInputErrors([]);
+      }
+    }
```

Comment:
Just combine this into one, you don't need to nest here.  `if (ticketErrors?.length > 0)` is sufficient, or `if (ticketErrors && ticketErrors.length > 0)` if you like that better.

Reply:
Argument of type 'string[] | undefined' is not assignable to parameter of type 'SetStateAction<string[]>'.
  Type 'undefined' is not assignable to type 'SetStateAction<string[]>'.

Reply:
I'm confused here, how are you getting that error if you do `if (ticketErrors && ticketErrors.length > 0)`, that should guarentee it can't be undefined.  what line are you getting it on.